### DISCUSSION TITLE
Can I upgrade crud from ^5.x.x to ^6.0.0 without surprises?

### DISCUSSION THREAD

**Discussion by theliuk** [mia-platform/community#289] [Q&A] [answered]
### Which section?

Console

### Which is the version?

11

### Description

Hi, we'd like to upgrade our crud-service to the new major, but we want to be sure not to break our architecture.
By looking at the ["breaking changes"](https://docs.mia-platform.eu/docs/runtime_suite/crud-service/changelog#breaking-changes) we don't see any serious breaking change.

Before upgrading we wanted to know if anybody experienced any issue in the process of migrating from ^5.x.x to ^6.x.x.

Thank you,

Luca

**Comment by fredmaggiowski** [comment]
Hey, CRUD Service v6 features a major update from ajv (it now uses [v8](https://github.com/ajv-validator/ajv/releases/tag/v8.0.0)), therefore there may be validation issues on complex schemas and some changes to the validation error messages the service may send back in case of a 400 status code.

Unless you have weird schemas or you rely on error messages for something the upgrade should be painless.

**Comment by theliuk** [comment]
Is it normal that crud service 6.5.0 takes a lot of time to start up? By "a lot of time" I mean w.r.t. the ^5.x.x.

**Reply by fredmaggiowski** [reply]
A slow down has been noticed with version 6.7.0 and we though it was due to an update to lc39 and fastify v4 🤔  

@danibix95 may provide further information

**Reply by danibix95** [reply, marked as answer]
Yep, we encountered an issue while upgrading `lc39`, but it has been resolved in version `v6.8.0`, which is already available and we recommend to upgrade to it. 

Nonetheless, we also revised the validation of the requests and responses, which at startup might require a bit more CPU usage, because the service has carry out more actions during the compilation of collection definitions schemas.

In addition, starting from `v6.8.0` there is an additional environment variable `ENABLE_STRICT_OUTPUT_VALIDATION` which allows to enable a stricter validation on CRUD Service response. This is an _opt-in_ feature, since enabling it might prevent the access to records of collections containing spurious data (document not totally conforming to the defined collection schema)

**Reply by theliuk** [reply]
Thank you @danibix95 .

@alexBonadei FYI
